root/static/scripts/guess-case/modes.js

```diff
@@ -282,3 +291,27 @@ exports.Sentence = assign({}, DefaultMode, {
     {url: {href: 'https://musicbrainz.org/doc/Style/Language', target: '_blank'}}
   ),
 });
+
+exports.Turkish = assign({}, DefaultMode, {
+  description: l(
+    'This mode capitalises almost all words, the connectives ' +
+    'and the question suffixes used in a title are lowercased.'+
+    'In Turkish, capital of \'i\' is \'İ\' and of \'ı\' is \'I\' and'+
+    'this mode handles these. Some words, often proper nouns, ' +
+    'may need to be manually fixed according to the {url|Turkish ' +
+    'language guidelines}.',
```

Comment:
Now I notice another thing: “Some words, often proper nouns, …” – that’s not really true anymore, is it? As it is now, the mode is more likely to over-capitalize rarer connectors.

I think I would simply drop the “often proper nouns” part. (In addition to the other changes suggested by @CallerNo6, of course.)